types/header.go

```diff
@@ -0,0 +1,193 @@
+	ProposerAddress []byte // original proposer of the block
+
+	// Hash of block aggregator set, at a time of block creation
+	AggregatorsHash [32]byte
+}
+
+func (h *Header) New() header.Header {
+	return new(Header)
+}
+
+func (h *Header) ChainID() string {
+	return h.BaseHeader.ChainID
+}
+
+func (h *Header) Height() int64 {
+	return int64(h.BaseHeader.Height)
+}
+
+func (h *Header) LastHeader() header.Hash {
+	return h.LastHeaderHash[:]
+}
+
+func (h *Header) Time() time.Time {
+	return time.Unix(int64(h.BaseHeader.Time), 0)
+}
+
+func (h *Header) IsRecent(blockTime time.Duration) bool {
+	return time.Since(h.Time()) <= blockTime
+}
+
+func (h *Header) IsExpired() bool {
+	// TODO(tzdybal): TrustingPeriod will be configurable soon (https://github.com/celestiaorg/celestia-node/pull/1544)
+	const TrustingPeriod = 168 * time.Hour
+	expirationTime := h.Time().Add(TrustingPeriod)
+	return !expirationTime.After(time.Now())
+}
+
+func (h *Header) VerifyAdjacent(untrst header.Header) error {
+	untrstH, ok := untrst.(*Header)
+	if !ok {
+		return &header.VerifyError{
+			Reason: fmt.Errorf("%T is not of type %T", untrst, h),
+		}
+	}
+
+	if untrstH.Height() != h.Height()+1 {
+		return &header.VerifyError{
+			Reason: fmt.Errorf("headers must be adjacent in height: trusted %d, untrusted %d", h.Height(), untrstH.Height()),
+		}
+	}
+
+	if err := verifyNewHeaderAndVals(h, untrstH); err != nil {
+		return &header.VerifyError{Reason: err}
+	}
+
+	// Check the validator hashes are the same
+	// TODO: next validator set is not available
+	if !bytes.Equal(untrstH.AggregatorsHash[:], h.AggregatorsHash[:]) {
+		return &header.VerifyError{
+			Reason: fmt.Errorf("expected old header next validators (%X) to match those from new header (%X)",
+				h.AggregatorsHash,
+				untrstH.AggregatorsHash,
+			),
+		}
+	}
+
+	return nil
+
+}
+
+func (h *Header) VerifyNonAdjacent(untrst header.Header) error {
+	untrstH, ok := untrst.(*Header)
+	if !ok {
+		return &header.VerifyError{
+			Reason: fmt.Errorf("%T is not of type %T", untrst, h),
+		}
+	}
+	if untrstH.Height() == h.Height()+1 {
+		return &header.VerifyError{
+			Reason: fmt.Errorf(
+				"headers must be non adjacent in height: trusted %d, untrusted %d",
+				h.Height(),
+				untrstH.Height(),
+			),
+		}
+	}
+
+	if err := verifyNewHeaderAndVals(h, untrstH); err != nil {
+		return &header.VerifyError{Reason: err}
+	}
+
+	// Ensure that untrusted commit has enough of trusted commit's power.
+	// err := h.ValidatorSet.VerifyCommitLightTrusting(eh.ChainID, untrst.Commit, light.DefaultTrustLevel)
+	// if err != nil {
+	// 	return &VerifyError{err}
+	// }
+
+	return nil
+}
+
+func (h *Header) Verify(h2 header.Header) error {
+	// TODO(tzdybal): deprecated
+	panic("implement me")
+}
+
+func (h *Header) Validate() error {
+	return h.ValidateBasic()
+}
+
+// clockDrift defines how much new header's time can drift into
+// the future relative to the now time during verification.
+var maxClockDrift = 10 * time.Second
```

Comment:
This will be removed after updating go-header - IsRecent / IsExpired methods are will not be a part of header interface: see https://github.com/celestiaorg/celestia-node/pull/1304/commits/0168021ffe171766d8567b802ef0ce923f90efc7